### PR TITLE
Added Data Parallelism tutorial in blitz

### PR DESCRIPTION
**Pull request by hunkim**
Data Parallelism tutorial is added.
- Changed the narrative style
- Added blitz (where is the optional step?)

Please take a look and give me lots of comments. @chsasank, CC: @soumith

### CODE REVIEW

**Comment by chsasank** [comment]
Thanks, this looks great. 

> Added blitz (where is the optional step?)

Can you edit last few lines of cifar10_tutorial so that the readers know this is an optional next step?
It'll also be nice if title of this tutorial has 'Optional: ' prefix.

**Comment by hunkim** [comment]
@chsasank Thanks! Changed the name to `Optional: Data Parallelism`

Added in the cifar10 tutorial: If you want to see even more MASSIVE speedup using all of your GPUs, please check out Optional: Data Parallelism.

File: beginner_source/blitz/data_parallel_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+# 
+
+
+######################################################################
+# Summary
+# -------
+# 
+# DataParallel splits your data automatically and sends job orders to multiple
+# models on several GPUs. After each model finishes their job, DataParallel
+# collects and merges the results before returning it to you. It's really awesome!
```
Comment (chsasank) [inline beginner_source/blitz/data_parallel_tutorial.py]: Can you remove it's awesome line? You know, it doesn't fit to the tone, I think.

File: beginner_source/blitz/data_parallel_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
+"""
+Optional: Data Parallelism
+============
```
Comment (chsasank) [inline beginner_source/blitz/data_parallel_tutorial.py]: I see ` WARNING: Title underline too short.` during sphinx build. Can you fix all such warnings?

**Comment by chsasank** [comment]
Thanks, merged!

**Pull request merged**